leases/utils.py

```diff
@@ -43,6 +45,58 @@ def calculate_season_end_date(lease_end: date = None) -> date:
     return default
 
 
+def calculate_winter_season_start_date(lease_start: date = None) -> date:
+    """Return the date when the winter season starts
+
+    If the current date is after the season end date,
+    it returns next year's default start date.
+
+    Winter Storage Leases always start on 15.9 the earliest
+    """
+    today = date.today()
+    default = date(day=15, month=9, year=today.year)
+
+    if lease_start:
+        lease_date = default.replace(year=lease_start.year)
+        # If the lease started between 1.1 and 10.6, that means
+        # the season began the previous year
+        if lease_start <= date(day=10, month=6, year=lease_start.year):
+            lease_date = default.replace(year=lease_start.year - 1)
```

Comment:
total nitpick, but would `lease_date = -= relativedelta(years=1)` work here and make `calculate_winter_season_start_date` more consistent? 😅